lib/lattice_field.cpp

```diff
@@ -461,11 +488,15 @@ namespace quda {
         const int num_dir = (comm_dim(dim) == 2 && comm_peer2peer_enabled(0,dim) && comm_peer2peer_enabled(1,dim)) ? 1 : 2;
 	for (int dir=0; dir<num_dir; ++dir) {
 	  if (!comm_peer2peer_enabled(dir,dim)) continue;
-	  void **ghostDest = &(ghost_remote_send_buffer_d[b][dim][dir]);
-	  cudaIpcOpenMemHandle(ghostDest, ipcRemoteGhostDestHandle[b][dir][dim],
-			       cudaIpcMemLazyEnablePeerAccess);
-	}
-	if (num_dir == 1) ghost_remote_send_buffer_d[b][dim][1] = ghost_remote_send_buffer_d[b][dim][0];
+#ifndef NVSHMEM_COMMS
+          void **ghostDest = &(ghost_remote_send_buffer_d[b][dim][dir]);
+          cudaIpcOpenMemHandle(ghostDest, ipcRemoteGhostDestHandle[b][dir][dim], cudaIpcMemLazyEnablePeerAccess);
+#else
+          ghost_remote_send_buffer_d[b][dim][dir]
+            = nvshmem_ptr(static_cast<char *>(ghost_recv_buffer_d[b]), comm_neighbor_rank(dir, dim));
```

Comment:
We need to re-construct the pointer whenever we change the communicator - the same peer2peer handles need to be refreshed.

Reply:
Yes, but we might just be fine here. Will need to check that later.

Reply:
We can probably address this in #1091.